OpenRA.Mods.Common/Orders/DemolitionOrderTargeter.cs

```diff
@@ -0,0 +1,36 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2016 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Orders
+{
+	class DemolitionOrderTargeter : UnitOrderTargeter
```

Comment:
If you end up being able to actually unify some more of the weird OTs, I could get behind this. If it's just taking an implementation detail of a trait and throwing it somewhere else, I don't think it's a good plan.